### ISSUE TITLE
Inaccuracy in COB leading to a possible insulin overdose.

### ISSUE THREAD

**Issue by bchris21** [nightscout/AndroidAPS#631]
Hello,

I am using NSClient to monitor my son while he is in kindergarten. I have realized the following issue which I would like to report:

**Issue:**
COB between AAPS and NSClient are inaccurate and not synced.

When does it happen:
Whenever I manually delete carbs in AAPS from Treatments tab. The carbs deletion does not sync in NSclient and COB is different including the extra carbs added.

**Why it can be dangerous:**
When my son is in kindergarten and I use Bolus Calculator (in NSclient) before sending SMS bolus, NSClient miscalculates bolus as it counts more carbs on board thus suggests more insulin and risking an overdose.

**Workaround:**
I have to manually check deleted carb entries in AAPS and manually remove them from NSClient Treatments' tab.

**App Versions/Builds:**
AAPS 2.8.2 Build: c07465b0c-2021-03.07-00:56 Fullrelease
NSClient 2.8.2 Build: 194549adc-2021-01-23-13:06 nsclientfullrelease

**Attachments:**
![Screenshot_20210814_131553](https://user-images.githubusercontent.com/4725280/129444617-2f2d844b-493a-47fc-a51d-2b50e4240ce6.jpg)
![Screenshot_20210814_131459](https://user-images.githubusercontent.com/4725280/129444619-2643bcba-f5ef-4911-b196-19acf6e448ab.jpg)
![Screenshot_20210814_131416](https://user-images.githubusercontent.com/4725280/129444620-bc509ad0-167b-4413-819c-8a863d1e3796.jpg)
![Screenshot_20210814_131347](https://user-images.githubusercontent.com/4725280/129444621-d1d5c8d4-0572-4ba5-8afc-e1ec5417ee6a.jpg)

I replicated the issue at 14/08/21 13:20 CET 

Logs from both AAPS and NSClient are attached.

[AndroidAPS_LOG_1628940043853.log.zip](https://github.com/nightscout/AndroidAPS/files/6986371/AndroidAPS_LOG_1628940043853.log.zip)
[AndroidAPS_LOG_1628940082178.log.zip](https://github.com/nightscout/AndroidAPS/files/6986372/AndroidAPS_LOG_1628940082178.log.zip)

Any help would be appreciated.

Thanks, 

Chris

**Comment by MilosKozak** [comment]
can you send screenshots with COB graphs when you see this issue again?

**Comment by bchris21** [comment]
@MilosKozak below the Screenshots requested.

**In detail:**
- I added fake 10gr at 23:57 to replicate the issue.
![Screenshot_20210818_000354_info.nightscout.nsclient.jpg](https://user-images.githubusercontent.com/4725280/129807580-644821ca-467f-439f-9c7d-c7b72650db51.jpg)

- I waited 5mins for the COB graph to update and took screenshots on both apps.
_AAPS Screenshot_
![Screenshot_20210818_000610.jpg](https://user-images.githubusercontent.com/4725280/129807772-97d31231-62f3-47ac-8f8c-974ce57ee674.jpg)

_NSClient Screenshot_
![Screenshot_20210818_000657.jpg](https://user-images.githubusercontent.com/4725280/129807799-4c981b66-f376-4ee4-8465-792a530cfe87.jpg)

- Then I removed the carbs from AAPS's treatment tab but still carbs are shown on NSClient and started getting "consumed".

_AAPS Screenshot_
![Screenshot_20210818_000845.jpg](https://user-images.githubusercontent.com/4725280/129807854-b9a4f784-6ed4-42aa-a1bb-b93a062dcfc3.jpg)

_NSClient Screenshot_
![Screenshot_20210818_000746.jpg](https://user-images.githubusercontent.com/4725280/129807891-0da4e650-f870-4ab3-95dc-cab033f24b57.jpg)

**Comment by MilosKozak** [comment]
so the problem is only when you you delete carbs?

**Comment by bchris21** [comment]
Exactly. Carbs addition syncs ok, but once carbs are deleted they do not sync on NSClient and they are accumulating until consumed by the pass of time or by BG drop.

**Comment by MilosKozak** [comment]
is the entry removed from NS?

**Comment by bchris21** [comment]
Hi Milos, sorry for the late reply. Just replicated the issue and I can confirm that carbs are removed from NS on the next BG update.
